### PR TITLE
[typescript] Annotation syntax addition to grammar

### PR DESCRIPTION
**Pull request by maikelsteneker**
This error occurs when executing CPD on a TypeScript file that includes annotations:

> The following error(s) occurred while 'CPD' was running: 'Lexical error in file [file] at line 15, column 1. Encountered: "@" (64), after : ""'.

This pull request adds support for this.

### CODE REVIEW

**Comment by jsotuyod** [comment]
@maikelsteneker thanks for the PR!

Adding [support for TypeScript](https://github.com/pmd/pmd/wiki/Project-Ideas-%5BInception%5D#add-support-for-typescript) is in our roadmap, however, I don't feel this is the right approach.

TypeScript is a different language to JavaScript / EcmaScript. Altering the ES5 grammar to allow annotations as suggested has some imporant downsides:

* Illegal JavaScript identifiers would be accepted, even in js files (not ts).
* Annotations should be treated differently, not as any other identifier, so it would require to be a different token type. For instance, in JS it is illegal to write 2 identifiers next to each other (`oneVar twoVar`), while in TS *annotations* may precede an identifier (`@validate greet()`)

Bottom line is, I believe a new module should be added for TypeScript, with a full proper grammar; where annotations are an entity themselves, and the grammar is explicit as to where they may appear or not.
